src/FlywheelCore.sol

```diff
@@ -164,7 +164,7 @@ contract FlywheelCore is Auth {
     /// @notice swap out the flywheel rewards contract
     function setFlywheelRewards(IFlywheelRewards newFlywheelRewards) external requiresAuth {
         uint256 oldRewardBalance = rewardToken.balanceOf(address(flywheelRewards));
-        if (oldRewardBalance > 0) {
+        if (address(flywheelRewards) != address(0) && oldRewardBalance > 0) {
```

Comment:
would be more gas efficient to not make the balance check if flywheelRewards = 0, so can we move this check up outside the transfer segment

Reply:
Good call Joey, just pushed these changes:
- Update the condition to only check whether `flywheelRewards` is the zero address
- Call `balanceOf` for the reward token on the prev. flywheel rewards contract
- Check if the balance is non-zero and, if true, transfer the balance to the new flywheel rewards before updating

Reply:
Just added a new check to prevent subsequent Flywheel Reward contracts from being the zero address (otherwise, old balances might be lost). I think it's fine since `setFlywheelRewards` should be called infrequently, but would love to hear your thoughts @Joeysantoro .

Reply:
I like it, good check to include when token transfers are involved